Assignment4/gor2000-a180374/task07.py

```diff
@@ -0,0 +1,92 @@
+# Assigment 4 -> task 07
+
+__author__ = "Jorge Garcia Martin"
+__email__ = "jorge.garciama@alumnos.upm.es"
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2020-2021/master/Assignment4"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage + "/resources/example6.rdf", format="xml")
+
+# **TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**
+
+# 7.1 RDFLib
+VCARD = 'http://www.w3.org/2001/vcard-rdf/3.0#'
+ns = Namespace("http://somewhere#")
+
+print("7.1 RDFLib:")
+for s, p, o in g.triples((None, RDFS.subClassOf, ns.Person)):
+    print(s, p, o)
+    for x, y, z in g.triples((None, RDFS.subClassOf, s)):
+        print(x, y, z)
+
+print("7.1 SPARQL:")
+from rdflib.plugins.sparql import prepareQuery
+
+sql1 = '''
+    SELECT DISTINCT ?x WHERE
+    { ?x rdfs:subClassOf* ns:Person }
+    LIMIT 10
```

Comment:
Limiting to 10 could mean that you are not obtaining all results, although it's ok in this case